### ISSUE TITLE
Unable to compile for TL-TW841N v11

### ISSUE THREAD

**Issue by chankruze** [ffulm/firmware#21]
I always facing this issue..
`make -r world: build failed. Please re-run make with -j1 V=s to see what's going on
make: *** [world] Error 1
`
I re-run make with -j1 V=s and caught the following error....
```
....
....
I: stripping and copying dynamic linker.
mips-openwrt-linux-musl-objcopy: '/home/chankruze/openwrt/tmp/mklibs-out/ld-musl-mips-sf.so.1': No such file
Command failed with status 1 : mips-openwrt-linux-musl-objcopy --strip-unneeded -R .note -R .comment  /home/chankruze/openwrt/tmp/mklibs-out/ld-musl-mips-sf.so.1
With output: 
make[2]: *** [package/install] Error 1
make[2]: Leaving directory `/home/chankruze/openwrt'
make[1]: *** [/home/chankruze/openwrt/staging_dir/target-mips_24kc_musl/stamp/.package_install] Error 2
make[1]: Leaving directory `/home/chankruze/openwrt'
make: *** [world] Error 2
chankruze@resonance-2:~/openwrt$ 

```
Full log files are attached.

- [fail-001-short.txt](https://github.com/ffulm/firmware/files/2476041/fail-001-min.txt)
- [fail-001-long.txt](https://github.com/ffulm/firmware/files/2476042/fail-001-long.txt)

**Comment by chankruze** [comment]
2nd line of above comment : **mips-openwrt-linux-musl-objcopy: '/home/chankruze/openwrt/tmp/mklibs-out/ld-musl-mips-sf.so.1': No such file** 
is the error i guess , due to which make is failing.

**Comment by chankruze** [comment]
@mwarning this is more important then #22.Because of this i am unable to compile.Planning to try to fix this...let's see what happens.

**Edit**:
Still same error-  **ld-musl-mips-sf.so.1** not found.
```
mkdir -p /home/chankruze/openwrt/tmp/mklibs-out
/home/chankruze/openwrt/staging_dir/host/bin/mklibs -D -d /home/chankruze/openwrt/tmp/mklibs-out --sysroot /home/chankruze/openwrt/staging_dir/target-mips_24kc_musl/root-ar71xx `cat /home/chankruze/openwrt/tmp/mklibs-libs | sed 's:/*[^/]\+/*$::' | uniq | sed 's:^/home/chankruze/openwrt/staging_dir/target-mips_24kc_musl/root-ar71xx:-L :'` --ldlib /lib/ld-musl-mips-sf.so.1 --target mips-openwrt-linux-musl `cat /home/chankruze/openwrt/tmp/mklibs-progs /home/chankruze/openwrt/tmp/mklibs-libs` 2>&1
I: Using /home/chankruze/openwrt/staging_dir/target-mips_24kc_musl/root-ar71xx/lib/ld-musl-mips-sf.so.1 as dynamic linker.
I: library reduction pass 1
0 symbols, 0 unresolved
I: stripping and copying dynamic linker.
mips-openwrt-linux-musl-objcopy: '/home/chankruze/openwrt/tmp/mklibs-out/ld-musl-mips-sf.so.1': No such file
Command failed with status 1 : mips-openwrt-linux-musl-objcopy --strip-unneeded -R .note -R .comment  /home/chankruze/openwrt/tmp/mklibs-out/ld-musl-mips-sf.so.1
With output: 
make[2]: *** [package/install] Error 1
make[2]: Leaving directory `/home/chankruze/openwrt'
make[1]: *** [/home/chankruze/openwrt/staging_dir/target-mips_24kc_musl/stamp/.package_install] Error 2
make[1]: Leaving directory `/home/chankruze/openwrt'
make: *** [world] Error 2
chankruze@resonance-2:~/openwrt$ 

```

**Comment by mwarning** [comment]
Building worked for me on Arch Linux (and after reverting the commits for cmake until 3.10.1).
But on Debian everything should compile out of the box.

Have you tried to compile (Vanilla) OpenWrt?
```
git clone https://git.openwrt.org/openwrt/openwrt.git
cd openwrt

./scripts/feeds update -a
./scripts/feeds install -a

make menuconfig
```

**Comment by chankruze** [comment]
Yes ...  openwrt is compiling without any errors... but not freifunk .
What i mean is without cloning freifun firmware and patches ... it is compiling with no errors for openwrt only.

**Comment by chankruze** [comment]
And one more querry ... i am getting `xyz.squashfs` file compiled (when compile Vanilla)but how can i flash this ? Do i need to use other tools to build `firmware-xyz.bin` ?
From menuconfig i can also get filesystem archive , i will try building from that.

**Comment by chankruze** [comment]
@mwarning Hey,
I am now able to compile vanilla (openwrt) without any patches which is now building **-factory.bin** and  **-sysupgrade.bin** :+1: 

The issue was i was including some more packages due to which it was exceeding the flash size 4MiB.So with default defconfig it is building all images but one issue it no web interface so i am now going to compile including luci core.let's see what happen !

**Comment by mwarning** [comment]
Current OpenWrt is at its limits with 4mb flash devices. The prime problem is the linux kernel that has grown in size over the years. LuCI (the default web ui) with OpenWrt alone might even exceed the 4mb threshold.

**Comment by mwarning** [comment]
Did the error message you mentioned come from exceeding the size limit? If so, then we can close this ticket. There would be nothing that can be done on our part.

**Comment by chankruze** [comment]
No this(exceeding size limit) is not responsible for this issue.This is error- **ld-musl-mips-sf.so.1 not found** and i have no idea what is going on.
Let me try again ! without the routing patch.
 
**Edit**:
I followed exactly what is in README and with routing patch also (because that bad commit hash was fixed) and this time it was successful.
So i am closing this issue.
